Mathlib/Algebra/GradedMonoid.lean

```diff
@@ -491,52 +490,66 @@ section Subobjects
-    [SetLike.HasGradedOne A] : ↑(@GradedMonoid.GhasOne.one _ (fun i => A i) _ _) = (1 : R) :=
+theorem SetLike.coe_gOne {S : Type _} [SetLike S R] [One R] [Zero ι] (A : ι → S)
+    [SetLike.GradedOne A] : ↑(@GradedMonoid.GOne.one _ (fun i => A i) _ _) = (1 : R) :=
   rfl
-#align set_like.coe_ghas_one SetLike.coe_ghas_one
+#align set_like.coe_ghas_one SetLike.coe_gOne
 
-/-- A version of `graded_monoid.ghas_one` for internally graded objects. -/
-class SetLike.HasGradedMul {S : Type _} [SetLike S R] [Mul R] [Add ι] (A : ι → S) : Prop where
+/-- A version of `GradedMonoid.ghas_one` for internally graded objects. -/
+class SetLike.GradedMul {S : Type _} [SetLike S R] [Mul R] [Add ι] (A : ι → S) : Prop where
+  /-- Multiplication is homoegenous -/
   mul_mem : ∀ ⦃i j⦄ {gi gj}, gi ∈ A i → gj ∈ A j → gi * gj ∈ A (i + j)
-#align set_like.has_graded_mul SetLike.HasGradedMul
+#align set_like.has_graded_mul SetLike.GradedMul
 
 theorem SetLike.mul_mem_graded {S : Type _} [SetLike S R] [Mul R] [Add ι] {A : ι → S}
-    [SetLike.HasGradedMul A] ⦃i j⦄ {gi gj} (hi : gi ∈ A i) (hj : gj ∈ A j) : gi * gj ∈ A (i + j) :=
-  SetLike.HasGradedMul.mul_mem hi hj
+    [SetLike.GradedMul A] ⦃i j⦄ {gi gj} (hi : gi ∈ A i) (hj : gj ∈ A j) : gi * gj ∈ A (i + j) :=
+  SetLike.GradedMul.mul_mem hi hj
 #align set_like.mul_mem_graded SetLike.mul_mem_graded
 
-instance SetLike.ghasMul {S : Type _} [SetLike S R] [Mul R] [Add ι] (A : ι → S)
-    [SetLike.HasGradedMul A] : GradedMonoid.GhasMul fun i => A i
-    where mul i j a b := ⟨(a * b : R), SetLike.mul_mem_graded a.Prop b.Prop⟩
-#align set_like.ghas_mul SetLike.ghasMul
+instance SetLike.gMul {S : Type _} [SetLike S R] [Mul R] [Add ι] (A : ι → S)
+    [SetLike.GradedMul A] : GradedMonoid.GMul fun i => A i
+    where mul := fun a b => ⟨(a * b : R), SetLike.mul_mem_graded a.prop b.prop⟩
+#align set_like.ghas_mul SetLike.gMul
 
-@[simp]
-theorem SetLike.coe_ghas_mul {S : Type _} [SetLike S R] [Mul R] [Add ι] (A : ι → S)
-    [SetLike.HasGradedMul A] {i j : ι} (x : A i) (y : A j) :
-    ↑(@GradedMonoid.GhasMul.mul _ (fun i => A i) _ _ _ _ x y) = (x * y : R) :=
+@[simp,nolint simpNF] 
+theorem SetLike.coe_gMul {S : Type _} [SetLike S R] [Mul R] [Add ι] (A : ι → S)
+    [SetLike.GradedMul A] {i j : ι} (x : A i) (y : A j) :
+    ↑(@GradedMonoid.GMul.mul _ (fun i => A i) _ _ _ _ x y) = (x * y : R) :=
   rfl
-#align set_like.coe_ghas_mul SetLike.coe_ghas_mul
+#align set_like.coe_ghas_mul SetLike.coe_gMul
+/-
+Porting note: Linter returns 
 
-/-- A version of `graded_monoid.gmonoid` for internally graded objects. -/
+"Left-hand side does not simplify, when using the simp lemma on itself."
+
+However, simp does indeed solve the following. Possibly related std#71,std#78
+
+example {S : Type _} [SetLike S R] [Mul R] [Add ι] (A : ι → S)
+    [SetLike.GradedMul A] {i j : ι} (x : A i) (y : A j) :
+    ↑(@GradedMonoid.GMul.mul _ (fun i => A i) _ _ _ _ x y) = (x * y : R) := by simp 
+
+-/
```

Comment:
Please move the porting note to just above where you added `nolint simpNF`. And please mention `simpNF` in the line that says `Porting note`. That way grepping for `Porting note` gives an indication what the note is about.

Reply:
Done!